uplc-environment.md

```diff
@@ -14,24 +14,22 @@ module UPLC-ENVIRONMENT
   imports BOOL
   imports INT-SYNTAX
   imports UPLC-MAP
+  imports UPLC-SYNTAX
   imports LIST
   imports K-EQUAL
 
-  syntax List ::= #append(List, Int) [function, functional]
-  rule #append(L:List, I:Int) => L ListItem(I)
-
   syntax Int ::= #last(List) [function]
   rule #last(L:List) => {L[-1]}:>Int
 
-  syntax Value ::= #lookup(Map, UplcId, Map) [function]
-  rule #lookup(E:Map, X:UplcId, H:Map) => {H[#last({E[X]}:>List)]}:>Value
-  requires X in_keys(E)
+  syntax Value ::= #lookup(Map, UplcId) [function]
+
+  // We could have a (spurious) case for when V is not a Value instead
+  // of that coercion, but perhaps that is handled by the new Env type?
+  rule [lookup.some]: #lookup(  E:Map,  X:UplcId) => { E[X] }:>Value
+  rule [lookup.none]: #lookup( _E:Map, _X:UplcId) => { < con integer 0 > }:>Value [owise]
```

Comment:
I think it might be better to leave the "none" case undefined and let it act as a failure point in llvm/haskell. Otherwise, the semantics could go down the "none" branch with `<con integer 0>` floating around in the configuration, and it wont be obvious if it belongs there or not when we're looking at it.

Reply:
Yes, since it will not be called given how `#lookup` is called in the first place.